core/templates/dev/head/pages/collection_editor/editor_tab/CollectionNodeCreatorDirective.js

```diff
@@ -76,6 +93,16 @@ oppia.directive('collectionNodeCreator', [
             );
           };
 
+          var convertTypeAheadToExplorationId = function(newExplorationId) {
+            if (newExplorationId.length > 1 && newExplorationId[0] === '(') {
+              var explorationId = '';
+              for (var i = 2; newExplorationId[i] !== ')'; i++) {
```

Comment:
Can't you use a regex and capture groups? Would be clearer, I think; this is pretty manual.